subctl/issue-182-subctl-a-kubectl-plugin.md

```diff
@@ -0,0 +1,149 @@
+This would be done by adding a new `make` target in `subctl` repository and running that target during the release process. Another option
+is to drop `.xz` extension files and only keep `.gz` extension files. The effect of this change to other projects would need to be
+considered, if any. Each file's sha256 checksum would be written to `subctl-checksums.txt`.
+
+The [root command name](https://github.com/submariner-io/subctl/blob/c3485569aa9e684dfc5ecad87668992cf377e1bc/cmd/subctl/root.go#L58) would
+need to be made variable based on the argument provided. This would allow running `kubectl-subm` command for the plugin and `subctl` command
+for standalone `subctl` binary.
+
+    // rootCmd represents the base command when called without any subcommands.
+    var rootCmd = &cobra.Command{
+    Use:   filepath.Base(os.Args[0]),
+    Short: "An installer for Submariner",
+    }
+
+Once `.tar.gz` files are available, krew plugin manifest file `subm.yaml` for subctl would be written as per
+[these instructions](https://krew.sigs.k8s.io/docs/developer-guide/plugin-manifest/#sample-plugin-manifest). The plugin name and krew
+manifest file's name must match. The manifest file will have an entry for each OS and architecture the tar file is generated for and its
+sha256. The manifest file would also mention the name and location of the `subctl` binary under the achieved folder and means to run it.
+
+Below is a sample `subm.yaml` manifest file.
+
+    apiVersion: krew.googlecontainertools.github.com/v1alpha2
+    kind: Plugin
+    metadata:
+      name: subm
+    spec:
+      version: "v0.15.0"
+      homepage: https://github.com/submariner-io/subctl
+      shortDescription: "Manages Submariner and its services"
+      description: |
+        CLI to install, uninstall and troubleshoot Submariner on a Kubernetes cluster.
+      platforms:
+        - selector:
+            matchLabels:
+              os: linux
+              arch: amd64
+          uri: https://github.com/submariner-io/releases/releases/download/v0.15.0/subctl-v0.15.0-linux-amd64.tar.gz
+          sha256: "eab60750423903167448c4888024336388592e867b95080aaf098bc670268399"
+          files:
+            - from: "subctl-*/subctl-*"
+              to: subctl
+          bin: subctl
+
+'files' lists which files should be extracted out from the `from` path under the downloaded archive to the `to` path. 'bin' specifies the
+path to the plugin executable among extracted files.
+
+Krew creates a symbolic link named `kubectl-subm`, derived from `metadata.name` field in the manifest file to `subctl` plugin executable
+after installation is complete.
+
+## Usage
+
+Once this is done, users would be able to install subctl using krew:
+
+    kubectl krew install subm
+    kubectl krew upgrade subm
+
+and use `subctl` as `kubectl-subm` or `kubectl subm`.
+
+    $ kubectl subm
+    An installer for Submariner
```

Comment:
Since subctl provides much more than installation, this should be changed to reflect that.  Maybe something like:

`A utility tool for Submariner`

Reply:
nit: I guess Vishal wanted the text to be updated accordingly.

Reply:
Yes, this text should also be change to make it clear that subctl is more than just an installer.